public/app/plugins/datasource/mysql/response_parser.ts

```diff
@@ -5,11 +5,6 @@ import { BackendDataSourceResponse, FetchResponse, toDataQueryResponse } from '@
 export default class ResponseParser {
   transformMetricFindResponse(raw: FetchResponse<BackendDataSourceResponse>): MetricFindValue[] {
     const frames = toDataQueryResponse(raw).data as DataFrame[];
-
-    if (!frames || !frames.length) {
```

Comment:
This code still exists for postgres and mssql. Should those also be changed? Is there any chance this and/or `transformAnnotationResponse` would fail in case of for example a 500 error, e.g. an error that only returns an error message from the backend and no data frames response at all?

Reply:
fixed the other return which doesn't contain dataframe when error. but to make sure that nothing could go wrong, and it is still making sense to check the length before taking any item from array, I would leave the frontend check.